helix-core/src/main/java/org/apache/helix/messaging/handling/HelixTaskExecutor.java

```diff
@@ -276,7 +276,7 @@ private void startMonitorThread() {
   /** Dedicated Thread pool can be provided in configuration or by client.
    *  This method is to check it and update the thread pool if necessary.
    */
-  private void updateStateTransitionMessageThreadPool(Message message, HelixManager manager) {
+  public void updateStateTransitionMessageThreadPool(Message message, HelixManager manager) {
```

Comment:
there is an annotation so that only it is visible for Test only.
can we use that, instead of making it public

Reply:
Yes, pls use @VisibleForTesting from Guava

Reply:
thanks for the review @desaikomal and @himanshukandwal , fixed it.

Reply:
Had to move the TestClass from **org.apache.helix** Package to **org.apache.helix.messaging.handling** Package to use annotations. Since there was another test class with the same name **TestHelixExecutor** in the package : org.apache.helix.messaging.handling, I ended up renaming this one to **TestHelixExecutor2**.